### ISSUE TITLE
communication lines not drawn correctly

### ISSUE THREAD

**Issue by JWegert** [ExplorViz/explorviz-ui-frontend#41]
The line (representing the communication between two classes) is not visualized correctly in the application level perspective. Right now it ends/starts above the class instances (see picture) , but the class instance should be the source/target point.
![timelinekommu](https://user-images.githubusercontent.com/17783114/35632109-04942baa-06a6-11e8-8a5d-249cc132c4b3.PNG)

**Comment by Alexander-Krause-Glau** [comment]
Temporary fixed with https://github.com/ExplorViz/explorviz-frontend/commit/03fd9f2bba8fc8997a3c8f10fb5a1b5570eabd93 (clipping problems still remain)

**Comment by malte-hansen** [comment]
@Alexander-Krause The clipping problems are not trivial to solve. I think it is a reasonable assumption that we can never avoid crossings of communication entirely since you can just add arbitrarily many communication lines (which then become hard to follow anyway).
We could do something to avoid clipping of communication and components. One approach would be to compute a route for communication which does not go through components but lies on top of comoponents like a loose string from one clazz to another. This approach would be in need of some heavy raycasting (for computing intersections) or some complex geometric calculations. 
Another approach, which I tried, is to use elliptic curves instead of lines. This leads to a result with practically no clipping issues. However, it makes communication hard to follow and looks more chaotic:

![curved communication](https://user-images.githubusercontent.com/38311128/56975620-7d596d80-6b71-11e9-8572-3bc73ee90c85.png)

**Comment by malte-hansen** [comment]
A possible solution would be to let the user decide whether he/she wants to use straight lines or curved lines for the visualization of communication. Additionally, the user should be able to configure the height of the curves. This could allow for both visually appealing and easy to follow communication lines (see figure below).
Since these properties should be persisted for multiple sessions, they should (also) be stored as user settings in the backend.

![curved communication low](https://user-images.githubusercontent.com/38311128/57216624-a43ff580-6ff0-11e9-971f-ed0a08fa18e1.png)

**Comment by Alexander-Krause-Glau** [comment]
TODO: Close when #160 was merged

**Comment by Alexander-Krause-Glau** [comment]
#160 was merged
